### PR TITLE
version 2 UI for seed restore

### LINKED ISSUE #505: Restore wallet UI: 2.0
https://gallery.io/projects/MCHbtQVoQ2HCZXGcfotOfnw9/files/MCEPul2DOrg5qGo5deUX_iuD

### LINKED ISSUE #549: Back up UI: Unable to select when there are many lines of prompts
You can choose success only with one line of prompt.

### PR DESCRIPTION
**Pull request by macsleven**
closes #505 
closes #549 

1. 

2. 

3. 

4. 

5. 

6.

### CODE REVIEW

**Comment by neddoz** [comment]
Please rebase this branch and also just noticed when you enter the words while recovering the wallet it does not automatically take use to password ntry automatically after the success of veryfying the words. I had to click that button.

File: Decred Wallet/Features/Wallet Setup/Recover Existing Wallet/RecoverExistingWalletViewController.swift [hunk 1]
```diff
@@ -101,37 +109,39 @@ class RecoverExistingWalletViewController: WalletSetupBaseViewController, UITabl
         let seedWordCell = tableView.dequeueReusableCell(withIdentifier: "seedWordCell", for: indexPath) as! RecoveryWalletSeedWordCell
         
         seedWordCell.lbSeedWordNum.text = String(format: LocalizedStrings.wordNumber, indexPath.row + 1)
+        seedWordCell.lbSeedWordNum.layer.borderColor = UIColor.appColors.darkBlue.cgColor
```
Comment (bgptr) [inline Decred Wallet/Features/Wallet Setup/Recover Existing Wallet/RecoverExistingWalletViewController.swift]:
```suggestion
        seedWordCell.lbSeedWordNum.layer.borderColor = UIColor.appColors.darkBluishGray.cgColor
```

File: Decred Wallet/Features/Wallet Setup/Recover Existing Wallet/RecoverExistingWalletViewController.swift [hunk 2]
```diff
@@ -101,37 +109,39 @@ class RecoverExistingWalletViewController: WalletSetupBaseViewController, UITabl
...
         seedWordCell.seedWordAutoComplete.text = self.userEnteredSeedWords[indexPath.row]
+        seedWordCell.cellBorder.layer.borderColor = UIColor.appColors.gray.cgColor
         seedWordCell.seedWordAutoComplete.resignFirstResponder()
         
         seedWordCell.setupAutoComplete(for: indexPath.row,
                                        filter: self.validSeedWords,
                                        dropDownListPlaceholder: self.wordSelectionDropDownContainer,
                                        onSeedEntered: self.seedWordEntered)
+        if indexPath.row == 0 {
+            seedWordCell.setRoundCorners(corners: [.topLeft, .topRight], radius: 10.0)
```
Comment (bgptr) [inline Decred Wallet/Features/Wallet Setup/Recover Existing Wallet/RecoverExistingWalletViewController.swift]:
```suggestion
            seedWordCell.setRoundCorners(corners: [.topLeft, .topRight], radius: 14.0)
```

File: Decred Wallet/Features/Wallet Setup/Recover Existing Wallet/RecoverExistingWalletViewController.swift [hunk 3]
```diff
@@ -101,37 +109,39 @@ class RecoverExistingWalletViewController: WalletSetupBaseViewController, UITabl
...
+        }
+        else if indexPath.row == 32 {
+            seedWordCell.setRoundCorners(corners: [.bottomRight, .bottomLeft], radius: 10.0)
```
Comment (bgptr) [inline Decred Wallet/Features/Wallet Setup/Recover Existing Wallet/RecoverExistingWalletViewController.swift]:
```suggestion
            seedWordCell.setRoundCorners(corners: [.bottomRight, .bottomLeft], radius: 14.0)
```

File: Decred Wallet/Extensions/UIViews.swift [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+//  UIView.swift
+//  Decred Wallet
+//
+// Copyright (c) 2018-2019 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+import UIKit
+
+extension UIView {
```
Comment (bgptr) [inline Decred Wallet/Extensions/UIViews.swift]: The other extension files' name is not plural

File: Decred Wallet/Extensions/UIViews.swift [hunk 5]
```diff
@@ -0,0 +1,30 @@
+//
+//  UIView.swift
+//  Decred Wallet
+//
+// Copyright (c) 2018-2019 The Decred developers
```
Comment (bgptr) [inline Decred Wallet/Extensions/UIViews.swift]:
```suggestion
// Copyright (c) 2019 The Decred developers
```

File: Decred Wallet/Features/Wallet Setup/Recover Existing Wallet/RecoverExistingWalletViewController.swift [hunk 6]
```diff
@@ -12,9 +12,8 @@ import Dcrlibwallet
 class RecoverExistingWalletViewController: WalletSetupBaseViewController, UITableViewDelegate, UITableViewDataSource {
     @IBOutlet var tableView : UITableView!
```
Comment (bgptr) [inline Decred Wallet/Features/Wallet Setup/Recover Existing Wallet/RecoverExistingWalletViewController.swift]:
```suggestion
    @IBOutlet var tableView: UITableView!
```

File: Decred Wallet/Features/Wallet Setup/Recover Existing Wallet/RecoverExistingWalletViewController.swift [hunk 7]
```diff
@@ -101,37 +116,53 @@ class RecoverExistingWalletViewController: WalletSetupBaseViewController, UITabl
...
         
         seedWordCell.setupAutoComplete(for: indexPath.row,
                                        filter: self.validSeedWords,
                                        dropDownListPlaceholder: self.wordSelectionDropDownContainer,
                                        onSeedEntered: self.seedWordEntered)
+        if indexPath.row == 0 {
+            seedWordCell.setRoundCorners(corners: [.topLeft, .topRight], radius: 14.0)
+            seedWordCell.cellComponentTopMargin.constant = 16
+        }
+        else if indexPath.row == 32 {
```
Comment (bgptr) [inline Decred Wallet/Features/Wallet Setup/Recover Existing Wallet/RecoverExistingWalletViewController.swift]:
```suggestion
        } else if indexPath.row == 32 {
```

File: Decred Wallet/Features/Wallet Setup/Recover Existing Wallet/RecoverExistingWalletViewController.swift [hunk 8]
```diff
@@ -101,37 +116,53 @@ class RecoverExistingWalletViewController: WalletSetupBaseViewController, UITabl
...
+            seedWordCell.setRoundCorners(corners: [.bottomRight, .bottomLeft], radius: 14.0)
+            seedWordCell.cellComponentBottomMargin.constant = 16
+        }
+        else{
```
Comment (bgptr) [inline Decred Wallet/Features/Wallet Setup/Recover Existing Wallet/RecoverExistingWalletViewController.swift]:
```suggestion
        } else {
```

File: Decred Wallet/Features/Wallet Setup/Recover Existing Wallet/RecoverExistingWalletViewController.swift [hunk 9]
```diff
@@ -101,37 +116,53 @@ class RecoverExistingWalletViewController: WalletSetupBaseViewController, UITabl
...
         
-        if (self.invalidSeedLabel.isHidden == false) {
-            self.invalidSeedLabel.isHidden = true
-            // increase top spacing so that confirm button is centered in display
-            self.tableViewFooterTopSpacingConstraint.constant = 30
-            UIView.animate(withDuration: 0.5) {
-                self.tableViewFooter.layoutIfNeeded()
-            }
+        switch indexPath.row {
+        case 0,32:
```
Comment (bgptr) [inline Decred Wallet/Features/Wallet Setup/Recover Existing Wallet/RecoverExistingWalletViewController.swift]:
```suggestion
        case 0, 32:
```

Comment (bgptr) [inline Decred Wallet/Features/Wallet Setup/Recover Existing Wallet/RecoverExistingWalletViewController.swift]: indentation

File: Decred Wallet/Features/Wallet Setup/WalletSetupViewController.swift [hunk 10]
```diff
@@ -13,8 +13,12 @@ class WalletSetupViewController: WalletSetupBaseViewController {
     override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
         if segue.identifier == "toNewWalletCreation" {
             Settings.setValue(true, for: Settings.Keys.NewWalletSetUp)
-        } else if segue.identifier == "toWalletRestore" {
-            Settings.setValue(false, for: Settings.Keys.NewWalletSetUp)
         }
     }
+    
+    @IBAction func RestoreWallet(_ sender: Any) {
```
Comment (bgptr) [inline Decred Wallet/Features/Wallet Setup/WalletSetupViewController.swift]:
```suggestion
    @IBAction func restoreWallet(_ sender: Any) {
```

File: Decred Wallet/Features/Wallet Setup/Recover Existing Wallet/RecoverExistingWalletViewController.swift [hunk 11]
```diff
@@ -40,12 +39,22 @@ class RecoverExistingWalletViewController: WalletSetupBaseViewController, UITabl
...
         self.wordSelectionDropDownContainer.layer.borderWidth = 1
         self.wordSelectionDropDownContainer.layer.borderColor = UIColor.appColors.lightGray.cgColor
+        self.wordSelectionDropDownContainer.setRoundCorners(corners: [.bottomRight, .bottomLeft, .topLeft, .topRight], radius: 4.0)
+        self.wordSelectionDropDownContainer?.layer.shadowColor = UIColor.appColors.darkBlue.cgColor
+        self.wordSelectionDropDownContainer?.layer.shadowRadius = 4
+        self.wordSelectionDropDownContainer?.layer.shadowOpacity = 0.24
+        self.wordSelectionDropDownContainer?.layer.shadowOffset = CGSize(width: -1, height: 1)
+        
+        // add drop shadow for better transition while scrolling the tableView
+        self.tableViewFooter.dropShadow(color: UIColor(hex: "#140000"), offSet: CGSize.zero )
```
Comment (bgptr) [inline Decred Wallet/Features/Wallet Setup/Recover Existing Wallet/RecoverExistingWalletViewController.swift]: color should come from `UIColor.appColors`

File: Decred Wallet/Features/Wallet Setup/Recover Existing Wallet/RecoverExistingWalletViewController.swift [hunk 12]
```diff
@@ -12,9 +12,8 @@ import Dcrlibwallet
 class RecoverExistingWalletViewController: WalletSetupBaseViewController, UITableViewDelegate, UITableViewDataSource {
     @IBOutlet var tableView : UITableView!
     @IBOutlet weak var wordSelectionDropDownContainer: UIView!
-    @IBOutlet weak var tableViewFooter: UIStackView!
-    @IBOutlet weak var tableViewFooterTopSpacingConstraint: NSLayoutConstraint!
-    @IBOutlet weak var invalidSeedLabel: UILabel!
+    @IBOutlet weak var tableViewFooterHeightCont: NSLayoutConstraint!
```
Comment (bgptr) [inline Decred Wallet/Features/Wallet Setup/Recover Existing Wallet/RecoverExistingWalletViewController.swift]:
```suggestion
    @IBOutlet weak var tableViewFooterHeightConst: NSLayoutConstraint!
```

File: Decred Wallet/Utils/Utils.swift [hunk 13]
```diff
@@ -10,6 +10,17 @@ import UIKit
 import JGProgressHUD
 import Dcrlibwallet
 
+enum BannerType: String {
+    case success
+    case error
+}
+
+struct AttributedStringStyle {
```
Comment (bgptr) [inline Decred Wallet/Utils/Utils.swift]: `AttributedStringStyle` struct does not need in this PR

**Pull request merged**